core-api/src/main/java/com/optimizely/ab/optimizelyjson/OptimizelyJSON.java

```diff
@@ -0,0 +1,153 @@
+/**
+ *
+ *    Copyright 2020, Optimizely and contributors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+package com.optimizely.ab.optimizelyjson;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
```

Comment:
We can't assume that `Gson` is available. I see two options:
1) We mimic the approach taken in [DefaultConfigParser](https://github.com/optimizely/java-sdk/blob/master/core-api/src/main/java/com/optimizely/ab/config/parser/DefaultConfigParser.java)
2) Take a stance on the serializer and include as a [shadow](https://imperceptiblethoughts.com/shadow/plugins/) dependency

Option 1 is the safer approach, since it has some precedence, but adds SDK complexity.
Option 2 is preferred but has a larger impact to the customer (my main concern is mobile). 

cc @mikeng13